source/parser/x86_64/allocators.hpp

```diff
@@ -53,35 +54,59 @@ namespace smeagle::x86_64 {
     }
 
     // Given two registers, return one combined string
-    std::string getRegistersString(std::pair<RegisterClass, RegisterClass> regClasses,
-                                   st::Type *paramType) {
-      std::string locA = this->getRegisterString(regClasses.first, paramType);
-      std::string locB = this->getRegisterString(regClasses.second, paramType);
-
-      // If B is empty (NO_CLASS) then return A
-      if (locB == "") {
-        return locA;
+    std::string getRegisterString(RegisterClass lo, RegisterClass hi, st::Type *paramType) {
+      if (lo == RegisterClass::NO_CLASS) {
+        throw std::runtime_error{"Can't allocate a {NO_CLASS, *}"};
+      }
+
+      if (lo == RegisterClass::MEMORY) {
+        // goes on the stack
+        return fallocator.nextFramebaseFromType(paramType);
       }
-      return locA + "|" + locB;
-    }
 
-    // Get a string location from a register class
-    std::string getRegisterString(RegisterClass regClass, st::Type *paramType) {
-      std::optional<std::string> regString;
+      if (lo == RegisterClass::INTEGER) {
+        auto reg = getNextIntRegister();
+        if (!reg) {
+          // Ran out of registers, put it on the stack
+          return fallocator.nextFramebaseFromType(paramType);
+        }
+        return reg.value();
+      }
 
-      // If the class is memory, pass the argument on the stack
-      if (regClass == RegisterClass::NO_CLASS) regString = "";
-      if (regClass == RegisterClass::SSE) regString = this->getNextSseRegister();
-      if (regClass == RegisterClass::INTEGER) regString = this->getNextIntRegister();
-      if (regClass == RegisterClass::MEMORY) regString = std::nullopt;
+      if (lo == RegisterClass::SSE) {
+        auto reg = getNextSseRegister();
+        if (!reg) {
+          // Ran out of registers, put it on the stack
+          return fallocator.nextFramebaseFromType(paramType);
+        }
+
+        if (hi == RegisterClass::SSEUP) {
+          // If the class is SSEUP, the eightbyte is passed in the next available eightbyte
+          // chunk of the last used vector register.
+        }
+        return reg.value();
+
+        /* TODO
+         *
+         *  For objects allocated in multiple registers, use the syntax '%r1 | %r2 | ...'
+         *  to denote this. This can only happen for aggregates.
+         *
+         *  Use ymm and zmm for larger vector types and check for aliasing
+         */
+      }
+
+      if (lo == RegisterClass::X87) {
+        // goes on the stack
+        return fallocator.nextFramebaseFromType(paramType);
+      }
 
-      // If we don't have a value, we need a framebase
-      if (!regString.has_value()) {
-        regString = fallocator.nextFramebaseFromType(paramType);
+      if (lo == RegisterClass::COMPLEX_X87) {
```

Comment:
Could this statement be put as an OR on line 98 (or do you anticipate more changes?)

Reply:
Ah, yes. That should have been merged together. I also missed the `X86UP` case there. I will fix it!